core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -64,7 +64,7 @@ private[spark] class SparkHadoopUtil extends Logging {
   }
 
   def createSparkUser(): UserGroupInformation = {
-    val user = Utils.getCurrentUserName()
+    val user = Option(System.getenv("HADOOP_USER_NAME")).getOrElse(Utils.getCurrentUserName())
```

Comment:
is "HADOOP_USER_NAME" constant defined?

Reply:
Yes, it's spec by spark.executorEnv.HADOOP_USER_NAME and all executors will use it